bundles/org.openhab.binding.linky/src/main/java/org/openhab/binding/linky/internal/handler/LinkyHandler.java

```diff
@@ -0,0 +1,387 @@
+                        logger.trace("Consumption at index {} added to current week: {}", jump, consumption);
+                    }
+                    yesterday = consumption;
+                }
+                jump++;
+                rangeStart = rangeStart.plusDays(1);
+            }
+        } else {
+            cachedDaylyData.invalidateValue();
+        }
+        updateKwhChannel(YESTERDAY, yesterday);
+        updateKwhChannel(THIS_WEEK, thisWeek);
+        updateKwhChannel(LAST_WEEK, lastWeek);
+    }
+
+    /**
+     * Request new monthly data and updates channels
+     */
+    private synchronized void updateMonthlyData() {
+        if (!isLinked(LAST_MONTH) && !isLinked(THIS_MONTH)) {
+            return;
+        }
+
+        Double lastMonth = Double.NaN;
+        Double thisMonth = Double.NaN;
+        LinkyConsumptionData result = cachedMonthlyData.getValue();
+        if (result != null && result.success()) {
+            int jump = result.getDecalage();
+            lastMonth = result.getData().get(jump).valeur;
+            thisMonth = result.getData().get(jump + 1).valeur;
+            if (thisMonth < 0) {
+                thisMonth = 0.0;
+            }
+        } else {
+            cachedMonthlyData.invalidateValue();
+        }
+        updateKwhChannel(LAST_MONTH, lastMonth);
+        updateKwhChannel(THIS_MONTH, thisMonth);
+    }
+
+    /**
+     * Request new yearly data and updates channels
+     */
+    private synchronized void updateYearlyData() {
+        if (!isLinked(LAST_YEAR) && !isLinked(THIS_YEAR)) {
+            return;
+        }
+
+        Double thisYear = Double.NaN;
+        Double lastYear = Double.NaN;
+        LinkyConsumptionData result = cachedYearlyData.getValue();
+        if (result != null && result.success()) {
+            int elementQuantity = result.getData().size();
+            thisYear = elementQuantity > 0 ? result.getData().get(elementQuantity - 1).valeur : Double.NaN;
+            lastYear = elementQuantity > 1 ? result.getData().get(elementQuantity - 2).valeur : Double.NaN;
+        } else {
+            cachedYearlyData.invalidateValue();
+        }
+        updateKwhChannel(LAST_YEAR, lastYear);
+        updateKwhChannel(THIS_YEAR, thisYear);
+    }
+
+    private void updateKwhChannel(String channelId, Double consumption) {
+        logger.debug("Update channel {} with {}", channelId, consumption);
+        updateState(channelId, consumption != Double.NaN ? new QuantityType<>(consumption, SmartHomeUnits.KILOWATT_HOUR)
+                : UnDefType.UNDEF);
+    }
+
+    /**
+     * Produce a report of all daily values between two dates
+     *
+     * @param startDay the start day of the report
+     * @param endDay the end day of the report
+     * @param separator the separator to be used betwwen the date and the value
+     *
+     * @return the report as a string
+     */
+    public String reportValues(LocalDate startDay, LocalDate endDay, @Nullable String separator) {
```

Comment:
From what I can tell, the only caller of this method splits apart the returned string into each line. Why can't this method just return a `List<String>`? Seems kinda strange to add in a bunch of newline characters only for callers of the method to immediately remove them.

It would be more efficient to pass in the List as a parameter so it can be used on recursive calls.
WDYT?

Reply:
I agree that it is a good remark.